alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -196,69 +158,113 @@ public void mapUsersToProjectByEmail(String projectHref, Collection<String> emai
     }
 
     @Override
-    public void deleteUsers(DescriptorKey descriptorKey, Collection<ProviderUserModel> users) {
-        users.forEach(user -> providerUserRepository.deleteByProviderAndEmailAddress(descriptorKey.getUniversalKey(), user.getEmailAddress()));
-    }
-
-    @Override
-    public List<ProviderUserModel> saveUsers(DescriptorKey descriptorKey, Collection<ProviderUserModel> users) {
-        List<ProviderUserModel> savedUsers = users
-                                                 .stream()
-                                                 .map(user -> convertToUserEntity(descriptorKey, user))
-                                                 .map(providerUserRepository::save)
-                                                 .map(entity -> convertToUserModel(entity))
-                                                 .collect(Collectors.toList());
-        return savedUsers;
+    public void updateProjectAndUserData(ProviderKey providerKey, Map<ProviderProject, Set<String>> projectToUserData) {
+        updateProjectAndUserData(providerKey, projectToUserData, Set.of());
     }
 
     @Override
-    public void updateProjectAndUserData(DescriptorKey descriptorKey, Map<ProviderProject, Set<String>> projectToUserData) {
-        updateProjectDB(descriptorKey, projectToUserData.keySet());
+    public void updateProjectAndUserData(ProviderKey providerKey, Map<ProviderProject, Set<String>> projectToUserData, Set<String> additionalRelevantUsers) {
+        updateProjectDB(providerKey, projectToUserData.keySet());
         Set<String> userData = projectToUserData.values().stream().flatMap(Collection::stream).collect(Collectors.toSet());
-        updateUserDB(descriptorKey, userData);
+        userData.addAll(additionalRelevantUsers);
+        updateUserDB(providerKey, userData);
         updateUserProjectRelations(projectToUserData);
     }
 
+    // ========================
+    // Non-interface public API
```

Comment:
These are used for the database subproject and tests.

Reply:
Looking at most of these methods, they are used within this class or strictly in tests. Are we sure we want to make them public? Generally we want to avoid methods specifically for testing.